test/runner/lib/cloud/vmware.py

```diff
@@ -0,0 +1,128 @@
+
+    def filter(self, targets, exclude):
+        """Filter out the cloud tests when the necessary config and resources
+        are not available.
+        :type targets: tuple[TestTarget]
+        :type exclude: list[str]
+        """
+
+        if os.path.isfile(self.config_static_path):
+            return
+
+        aci = self._create_ansible_core_ci()
+
+        if os.path.isfile(aci.ci_key):
+            return
+
+        if is_shippable():
+            return
+
+        super(VmwareCloudProvider, self).filter(targets, exclude)
+
+    def setup(self):
+        """Setup the cloud resource before delegation and register a
+        cleanup callback."""
+        super(VmwareCloudProvider, self).setup()
+
+        if not self._use_static_config():
+            self._setup_dynamic()
+
+    def cleanup(self):
+        """Clean up the cloud resource and any temporary configuration files
+        after tests complete."""
+        if self.aci:
+            self.aci.stop()
+        super(VmwareCloudProvider, self).cleanup()
+
+    def _setup_dynamic(self):
+        """Request Esxi credentials through the Ansible Core CI service."""
+        display.info('Provisioning %s cloud environment.' % self.platform,
+                     verbosity=1)
+
+        config = self._read_config_template()
+
+        aci = self._create_ansible_core_ci()
+
+        if not self.args.explain:
+            response = aci.start()
+
+        if not self.args.explain:
+            credentials = response['esxi']
+            self.aci = aci
+
+        if not self.args.explain:
+            values = dict(
+                VMWARE_HOST=credentials['VMWARE_HOST'],
+                VMWARE_USER=credentials['VMWARE_USER'],
+                VMWARE_PASSWORD=credentials['VMWARE_PASSWORD'],
+                VMWARE_HTTPS_PORT=str(credentials['VMWARE_HTTPS_PORT']),
+                VMWARE_SSH_PORT=str(credentials['VMWARE_SSH_PORT']),
+                VMWARE_VALIDATE_CERTS=credentials["VMWARE_VALIDATE_CERTS"],
+            )
+
+            config = self._populate_config_template(config, values)
+            self._write_config(config)
+
+    def _create_ansible_core_ci(self):
+        """
+        :rtype: AnsibleCoreCI
+        """
+        return AnsibleCoreCI(self.args, 'Vmware', 'Vmware',
+                             persist=False, stage=self.args.remote_stage,
+                             provider=self.args.remote_provider)
+
+
+class VmwareEnvironment(CloudEnvironment):
+    """VMware Vmware environment plugin. Updates integration test
+    environment after delegation."""
+    def get_environment_config(self):
+        """
+        :rtype: CloudEnvionmentConfig
+        """
+
+        parser = ConfigParser()
+        parser.read(self.config_path)
+
+        env_vars = {}
+
+        ansible_vars = dict(
+            resource_prefix=self.resource_prefix,
+        )
+        ansible_vars.update(dict(parser.items('DEFAULT')))
+
+        return CloudEnvironmentConfig(
```

Comment:
If we want to stay compatible with the current test-suite, we should expose the following Ansible variables:

- vcenter_hostname: VMWARE_HOST:VMWARE_PORT
- vcenter_username: VMWARE_USER
- vcenter_password: VMWARE_PASSWORD

We can also just exoise the VMWARE_* environment variables for this purpose. I like the idea since this would remove a lot of code. But in this case, we must do some extra adjustments:

- vcenter.py: it should export the same variables
- the test-suite, we should not use vcenter_* variable anymore.

Reply:
I don't really see that it removes any code in the provider?  We would just also export the config as env vars at https://github.com/ansible/ansible/blob/0d33aaa88df2bbcc840867b97870d67793e73d76/test/runner/lib/cloud/vmware.py#L118  
It ends up looking like the azure provider, for example.  If env vars are easier for the test targets though that totally works for me and is easy to implement.

Reply:
This is the kind of block that won't be necessary anymore:
https://github.com/ansible/ansible/blob/affaedfca8c14f70968a8d2a6c898c488adaa55a/test/integration/targets/vmware_vm_facts/tasks/main.yml#L16-L19

Reply:
Ah ok, I didn't realize you meant code in the tests - it is no longer needed with the most recent version then.
https://github.com/jillr/ansible/blob/vmware_provider/test/runner/lib/cloud/vmware.py#L105-L112